plugins/vpp/ifplugin/vppcalls/vpp1901/watch_vppcalls_test.go

```diff
@@ -0,0 +1,151 @@
+//  Copyright (c) 2019 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp1901_test
+
+import (
+	"testing"
+
+	"github.com/ligato/vpp-agent/plugins/vpp/binapi/vpp1901/interfaces"
+
+	"github.com/ligato/vpp-agent/plugins/vpp/binapi/vpp1901/dhcp"
+	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/vppcalls"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestWatchInterfaceEvents(t *testing.T) {
+	ctx, ifHandler := ifTestSetup(t)
+	defer ctx.TeardownTestCtx()
+	ctx.MockVpp.MockReply(&interfaces.WantInterfaceEventsReply{})
+	eventsChan := make(chan *vppcalls.InterfaceEvent)
+	err := ifHandler.WatchInterfaceEvents(eventsChan)
+	notifChan := ctx.MockChannel.GetChannel()
+	Expect(notifChan).ToNot(BeNil())
+	Expect(err).To(BeNil())
+
+	notifChan <- &interfaces.SwInterfaceEvent{
+		SwIfIndex:   1,
+		AdminUpDown: 1,
+		LinkUpDown:  1,
+		Deleted:     1,
+	}
+	var result *vppcalls.InterfaceEvent
+	checkIfState(1, result, eventsChan)
```

Comment:
Checking the expected results in `checkIfState` just creates unnecessary indirection and requires reader to jump up and down in code.

Try to contain tests inside single test function if possible. If you want to make it shorter you can use `Expect` to compare entire struct with values.